src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -103,8 +111,15 @@ export default class DefaultMessagingSession implements MessagingSession {
     });
   }
 
-  private startConnecting(reconnecting: boolean): void {
-    const signedUrl = this.prepareWebSocketUrl();
+  private async startConnecting(reconnecting: boolean) {
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : null;
+    if (endpointUrl == null) {
+      const endpoint = await this.configuration.chimeClient.getMessagingSessionEndpoint().promise();
```

Comment:
Does this syntax work with `aws-sdk-js` v3? We now support both aws-sdk v2 and v3 in `amazon-chime-sdk-js` v3 and the syntax seems different in v3:
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-chime-sdk-messaging/classes/getmessagingsessionendpointcommand.html

Reply:
Will update v3 but leave like it is here for v2